repose-aggregator/commons/utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -46,84 +53,253 @@ import javax.servlet.http.HttpServletResponse
+  override def setHeader(name: String, value: String): Unit = {
+    ifMutable(headerMode) {
+      headerMap = headerMap + (name -> Seq(value))
+    }
+  }
 
   /**
    * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
    */
-  override def replaceHeader(headerName: String, headerValue: String, quality: Double): Unit = ???
+  override def setIntHeader(name: String, value: Int): Unit = setHeader(name, value.toString)
 
-  override def getHeadersList(headerName: String): util.List[String] = ???
+  /**
+   * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
+   */
+  override def setDateHeader(name: String, timeSinceEpoch: Long): Unit =
+    setHeader(name, DateUtils.formatDate(new Date(timeSinceEpoch)))
 
-  override def getPreferredHeaders(headerName: String): util.List[String] = ???
+  /**
+   * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
+   */
+  override def removeHeader(name: String): Unit = {
+    ifMutable(headerMode) {
+      headerMap = headerMap - name
+    }
+  }
 
-  override def getPreferredSplittableHeaders(headerName: String): util.List[String] = ???
+  /**
+   * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
+   */
+  override def replaceHeader(name: String, value: String): Unit = setHeader(name, value)
 
-  override def getSplittableHeaders(headerName: String): util.List[String] = ???
+  /**
+   * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
+   */
+  override def replaceHeader(name: String, value: String, quality: Double): Unit = setHeader(name, s"$value;q=$quality")
 
   /**
    * @throws IllegalStateException when bodyMode is ResponseMode.PASSTHROUGH
    */
-  def getOutputStreamAsInputStream(): InputStream = ???
+  def getOutputStreamAsInputStream: InputStream = bodyOutputStream.getOutputStreamAsInputStream
 
   /**
    * @throws IllegalStateException when bodyMode is anything other than ResponseMode.MUTABLE
    */
-  def setOutput(inputStream: InputStream): Unit = ???
+  def setOutput(inputStream: InputStream): Unit = bodyOutputStream.setOutput(inputStream)
+
+  override def getWriter: PrintWriter = {
+    responseBodyType match {
+      case ResponseBodyType.OutputStream =>
+        throw new IllegalStateException("Cannot call getWriter after calling getOutputStream")
+      case _ =>
+        responseBodyType = ResponseBodyType.PrintWriter
+        bodyPrintWriter
+    }
+  }
+
+  override def getOutputStream: ServletOutputStream = {
+    responseBodyType match {
+      case ResponseBodyType.PrintWriter =>
+        throw new IllegalStateException("Cannot call getOutputStream after calling getWriter")
+      case _ =>
+        responseBodyType = ResponseBodyType.OutputStream
+        bodyOutputStream
+    }
+  }
+
+  override def flushBuffer(): Unit = {
```

Comment:
So the documentation says:
"Forces any content in the buffer to be written to the client. A call to this method automatically commits the response, meaning the status code and headers will be written."

That sounds a lot like a commit. However, if we do a full commit here, then the user who wrapped the request can not longer modify it. That seems contrary to the purpose of the wrapper.

I'm not exactly sure what you mean by counting on the underlying isCommitted.